internal/types/path.go

```diff
@@ -220,6 +220,13 @@ func insertByPath(doc *Document, path Path) error {
 						FormatAnyValue(v),
 					)
 				}
+			default:
+				return fmt.Errorf(
```

Comment:
Should this function be called for anything but doc/array at all?

Reply:
Please suggest where is a better to add this check :hugs:

I don't know the answer to _if this function should be called for non-doc/array_, but it _is_ being called for non-doc/array and we need to return error.

Reply:
Ok, I think I got the idea (sort of). I'll ask in person in a call not to extend the current thread.